application/src/test/java/uk/nhs/hee/tis/revalidation/validator/TraineeRecommendationRecordDTOValidatorTest.java

```diff
@@ -0,0 +1,235 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.validator;
+
+import static java.time.LocalDate.now;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationType.DEFER;
+
+import com.github.javafaker.Faker;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.Errors;
+import uk.nhs.hee.tis.revalidation.dto.TraineeRecommendationRecordDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.exception.RecommendationException;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+
+@ExtendWith(MockitoExtension.class)
+class TraineeRecommendationRecordDTOValidatorTest {
+
+  private final static LocalDate today = now();
+  private final Faker faker = new Faker();
+  private final LocalDate gmcSubmissionDate = now();
+  private final LocalDate deferralDate = now();
+  private final String gmcId = faker.number().digits(7);
+  private final String deferralReason1 = "1";
+  private final String deferralSubReason1 = "1";
+  private static final String DOCTOR_NOT_FOUND_MESSAGE = "Doctor %s does not exist!";
+  @Mock
+  private DoctorsForDBRepository doctorsForDBRepository;
+  @InjectMocks
+  private TraineeRecommendationRecordDTOValidator validatorUnderTest;
+  private DoctorsForDB doctorsForDB;
+
+
+  @Test
+  void shouldValidateWhenGmcNumberOrRecommendationTypeEmptyOrNullInRecommendationRequest() {
```

Comment:
This looks to be testing empty but not null.

Reply:
I have separted this method into two different methods
shouldValidateWhenGmcNumberOrRecommendationTypeIsEmptyInRecommendationRequest
shouldValidateWhenGmcNumberOrRecommendationTypeIsNullInRecommendationRequest